docs/DeveloperGuide.md

```diff
@@ -1,29 +1,739 @@
+| budget          | double        | The budget amount to be set for the category                        |
+
+##### Class Attributes for `ListBudgetCommand`:
+| Class Attribute | Variable Type | Relevance                                                           |
+|-----------------|---------------|---------------------------------------------------------------------|
+| expenseList     | ExpenseList   | Object containing the list of expenses to check against set budgets |
+
+
+Upon the call of the `execute()` method in `BudgetBuddy` using `command.execute()`, `SetBudgetCommand` will update the 
+budget in `ExpenseList` using `setBudget`. Similarly, `ListBudgetCommand` will fetch and display all categories with 
+their budgets using `getBudgets`, and highlight those that are above the set budget.
+
+##### Key Methods used from `ExpenseList`
+| Method                      | Return Type   | Relevance                                                          |
+|-----------------------------|---------------|--------------------------------------------------------------------|
+| setBudget(category, budget) | void          | Sets or updates the budget for a given category in the ExpenseList |
+| getBudgets()                | List<Budget>  | Retrieves the list of all budgets set                              |
+
+The `ListBudgetCommand`'s updated execution function now features an improved display that not only shows the budget, 
+spent amount, and remaining balance but also clearly indicates when the budget has been exceeded. If the expenses 
+surpass the budget, instead of showing a negative remaining balance, it displays "Exceeded", providing a straightforward
+and immediate visual cue that the budget limits have been surpassed.
+
+The "Categories above budget" section offers a concise table summarizing which categories have gone over the budget and
+by what amount, making it easy for users to identify areas of concern.
+
+
+#### Sequence diagrams
+
+##### Setting a Budget
+The following UML Sequence diagram shows how `SetBudgetCommand` works when a user sets a budget for a category in the
+following format: `set budget c/<Category> b/<Amount>`
+
+<img alt="SetBSD.png" height="400" src="SetBSD.png" width="700"/>
+
+
+##### Printing Budgets
+The following UML Sequence diagram shows how `ListBudgetCommand` works when a user checks the budget status with the
+command: `budget print`
+
+<img alt="ListBSD2.png" height="400" src="ListBSD2.png" width="700"/>
+
+#### Class diagram
+The class diagram below outlines the relationships between the classes involved in the Budget Management feature:
+
+<img alt="ClassDiagram2.png" height="800" src="ClassDiagram2.png" width="700"/>
+
+#### Activity diagram
+
+The activity diagram provides an overview of the Budget Management feature's workflow:
+
+<img alt="ActivityDiagram.png" height="600" src="ActivityDiagram.png" width="700"/>
```

Comment:
The workflow of Budget Management feature has been well explained here using the activity diagram, kudos!